### PR TITLE
Added draft neo4j-admin commands: setPassword, listTransactions

### PR DESCRIPTION
**Pull request by craigtaverner**
This is initial work, the set password command should work, but the list transactions is a stub.

### CODE REVIEW

File: community/security/src/main/java/org/neo4j/commandline/admin/security/SetPasswordCommand.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        @Override
+        public Optional<String> arguments()
+        {
+            return Optional.of( "<username> <password>" );
+        }
+
+        @Override
+        public String description()
+        {
+            return "Sets the password for the specified user and activates that user in case it is suspended";
```
Comment (OliviaYtterbrink) [inline community/security/src/main/java/org/neo4j/commandline/admin/security/SetPasswordCommand.java]: `activates that user` what does this mean? Are you referring to the fact that `password_change_required` is cleared?

Comment (craigtaverner) [inline community/security/src/main/java/org/neo4j/commandline/admin/security/SetPasswordCommand.java]: Good point. I should improve the message text.

File: community/security/src/test/java/org/neo4j/server/security/auth/AdminCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import org.neo4j.commandline.admin.security.SetPasswordCommand;
+import org.neo4j.logging.NullLogProvider;
+import org.neo4j.test.rule.TargetDirectory;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+
+public class AdminCommandTest
```
Comment (srbaker) [inline community/security/src/test/java/org/neo4j/server/security/auth/AdminCommandTest.java]: Should this be called `SetPasswordCommandTest` ?

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AdminCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import java.io.File;
+
+import org.neo4j.commandline.admin.security.ListTransactionsCommand;
+import org.neo4j.test.rule.TargetDirectory;
+
+public class AdminCommandTest
```
Comment (srbaker) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AdminCommandTest.java]: Should this be called `ListTransactionsCommandTest` ?

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AdminCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+        // When - the admin command sets the password
+        ListTransactionsCommand listTransactionsCommand = new ListTransactionsCommand( graphDir.toPath(), confDir.toPath() );
+        listTransactionsCommand.execute( new String[]{"--username=neo4j", "--password=abc"} );
+
+        // Then - it works
+        //TODO:
+    }
+
+}
```
Comment (srbaker) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AdminCommandTest.java]: Missing tests for the required arguments `username` and `password`.

File: community/security/src/test/java/org/neo4j/server/security/auth/AdminCommandTest.java (hunk 2 above)
Comment (craigtaverner) [inline community/security/src/test/java/org/neo4j/server/security/auth/AdminCommandTest.java]: I expected to add more tests for more commands, so kept the name generic. In the case of the enterprise tests, where will be more commands. In the case of community, perhaps we only have one.... in which case you might be right.

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AdminCommandTest.java (hunk 3 above)
Comment (craigtaverner) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AdminCommandTest.java]: Here we definitely expect to have a growing list of commands to test. I would prefer to keep these tests together.

File: community/security/src/test/java/org/neo4j/server/security/auth/AdminCommandTest.java (hunk 2 above)
Comment (chrisvest) [inline community/security/src/test/java/org/neo4j/server/security/auth/AdminCommandTest.java]: Security is not the only module that has admin commands. Perhaps call it SecurityAdminCommandsTest?

**Comment by craigtaverner** [comment]
We will split this into two PR's one for setpassword and one for listtransactions. See https://github.com/neo4j/neo4j/pull/7586 for setpassword.

**Pull request closed without merging**